### PR TITLE
Add screenreader-announced keyboard shortcuts for AdderToolbar

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR adds screen-reader only `aria-live` regions to the `AdderToolbar` to announce keyboard shortcuts for screen readers. The goal is to make these keyboard shortcuts discoverable to users of assistive technology without being too terribly annoying.

The implementation philosophy hews close to [documented intent and behavior of `aria-live` regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) without added cleverness. 

## Initially, these were the announcements

The intended announced content (and what I am observing in Chrome with VoiceOver on MacOS):

* The first time the adder toolbar is shown (i.e. the first time the user selects text in the guest page):
    * If there are (5, in these examples) associated annotations to show: _5 annotations for this selection. Press 's' to show annotations for the selected text. Press 'a' to create an annotation. Press 'h' to highlight._
    * If no annotations (this is the most common case): _Press 'a' to create an annotation. Press 'h' to highlight._
* Subsequent text selections (adder popups):
    * If there are no associated annotations with the selection: no announcement
    * If there are associated annotations with the selection:
        * If this is the first time there have been associated annotations with any selection (i.e. previous selections did not have any associated annotations): _5 annotations for this selection. Press 's' to show annotations for the selected text._
        * Else: _5 annotations for this selection._

## After some feedback, these are the current announcements

_Any_ time the adder is shown:

* If there are associated annotations for the selected text:  _5 annotations for this selection. Press 'S' to show annotations. Press 'A' to annotate. Press 'H' to highlight._
* If no associated annotations: _Press 'A' to annotate. Press 'H' to highlight._

We will want to get user feedback about whether these repeated announcements are troublesome.

Fixes https://github.com/hypothesis/client/issues/4742

### CODE REVIEW

**Comment by lyzadanger** [comment]
Argh. No good deed goes unpunished. Adding tests (at all) to `AdderToolbar-test` is causing test misses.

**Review by robertknight — COMMENTED** [review]
Some thoughts from my initial testing:

1. The basic mechanism seems to work well in VoiceOver and be a good screen-reader "version" of a toolbar. We need to test with other screen readers, especially on Windows.
2. I found that it was very easy to dismiss/skip over the "Press "a" to annotate, "h" to highlight" text that is read the first time, as if the user creates a selection and then immediately modifies it (eg. selects one word, then selects another) that will immediately dismiss it. A solution I tried locally was to modify the logic so that the prompt is read out once when the toolbar appears, but not again if the selection is modified. This gives the user a chance to hear it again if they miss it the first time, or just forget, but is hopefully not annoying as as the prompt is read after the selected text (we need to verify how this works in other systems)
3. I would suggest to shorten some of the statuses for efficiency of reading and also capitalize letters to ensure they are read as standalone letters rather than words ("Ayy" vs "ah")

File: src/annotator/components/AdderToolbar.tsx [hunk 1]
```diff
@@ -101,6 +101,47 @@ function ToolbarButton({
...
+            {annotationCount === 1 ? 'annotation' : 'annotations'} for this
+            selection.
+          </span>
+        )}
+      </span>
+      <ul aria-live="polite" data-testid="annotate-shortcuts-announce">
+        {annotationCount > 0 && (
+          <li>Press {"'s'"} to show annotations for the selected text.</li>
+        )}
+        <li>Press {"'a'"} to create an annotation.</li>
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/annotator/components/AdderToolbar.tsx:138]: This sounds good...

File: src/annotator/components/AdderToolbar.tsx [hunk 2]
```diff
@@ -101,6 +101,47 @@ function ToolbarButton({
...
+      <span
+        aria-live="polite"
+        aria-atomic="true"
+        role="status"
+        data-testid="annotation-count-announce"
+      >
+        {annotationCount > 0 && (
+          <span>
+            {annotationCount}{' '}
+            {annotationCount === 1 ? 'annotation' : 'annotations'} for this
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/annotator/components/AdderToolbar.tsx:129]: "x annotations for selection" sounds a little abrupt/strange to me.

**Comment by lyzadanger** [comment]
@robertknight 

> A solution I tried locally was to modify the logic so that the prompt is read out once when the toolbar appears, but not again if the selection is modified. 

Can you expand on this? I'm not quite grokking.

**Comment by lyzadanger** [comment]
@robertknight I've updated shortcut keys to be capitalized and shortened text as suggested. I'll wait to hear more from you on how you posit we might make the experience better if the user changes a selection right after making it...

**Comment by robertknight** [comment]
> Can you expand on this? I'm not quite grokking.

This diff is what I experimented with locally:

```diff
diff --git a/src/annotator/components/AdderToolbar.tsx b/src/annotator/components/AdderToolbar.tsx
index 44f1e3b68..9f724c4bd 100644
--- a/src/annotator/components/AdderToolbar.tsx
+++ b/src/annotator/components/AdderToolbar.tsx
@@ -112,8 +112,10 @@ function ToolbarButton({
  */
 function AdderToolbarShortcuts({
   annotationCount,
+  isVisible,
 }: {
   annotationCount: number;
+  isVisible: boolean;
 }) {
   return (
     <div className="sr-only">
@@ -132,11 +134,15 @@ function AdderToolbarShortcuts({
         )}
       </span>
       <ul aria-live="polite" data-testid="annotate-shortcuts-announce">
-        {annotationCount > 0 && (
-          <li>Press {"'s'"} to show annotations for the selected text.</li>
+        {isVisible && (
+          <>
+            {annotationCount > 0 && (
+              <li>Press S to show annotations.</li>
+            )}
+            <li>Press A to annotate.</li>
+            <li>Press H to highlight.</li>{' '}
+          </>
         )}
-        <li>Press {"'a'"} to create an annotation.</li>
-        <li>Press {"'h'"} to highlight.</li>
       </ul>
     </div>
   );
@@ -281,7 +287,10 @@ export default function AdderToolbar({
         )}
       </div>
       <AdderToolbarArrow arrowDirection={arrowDirection} />
-      <AdderToolbarShortcuts annotationCount={annotationCount} />
+      <AdderToolbarShortcuts
+        annotationCount={annotationCount}
+        isVisible={isVisible}
+      />
     </div>
   );
 }
```

**Comment by lyzadanger** [comment]
@robertknight I've updated to make shortcut announcements happen any time the adder is made visible. For the record, the reason that this works is that whenever the adder goes non-visible, the keyboard shortcuts are removed from the `ul[aria-live="polite"]` and then _re-added_ when the adder goes visible again. This `aria-live` region announces _additions_.

**Review by robertknight — APPROVED** [review]

**Pull request merged**